### PR TITLE
Add option to interact with most recent tab only.

### PR DESCRIPTION
**Pull request by metaflow**
See issue #337 and #322.

@berrberr Please suggest what tests you would like to see.

I also wonder if this should be under the option. I, personally, don't see the usecase when a user wants to start playing, toggle play/pause or select next song across all the open tabs.

### CODE REVIEW

**Comment by Wizek** [comment]
Wow, this is great! Thanks for implementing @metaflow! I intend to give this a try.

File: code/js/background.js [hunk 1]
```diff
@@ -17,21 +17,87 @@
...
+    var playing = getPlayingTabs(tabs);
+    if (_.isEmpty(playing)) {
+      sendActionAllPlayers(command, [getRecentTab(tabs)]);
+    } else {
+      sendActionAllPlayers(command, playing);
+    }
+  };
+
+  /**
+   * Returns most tab of most recently updated status.
```
Comment (nemchik) [inline code/js/background.js]: This comment doesn't read correctly.

Comment (metaflow) [reply code/js/background.js]: fixed

**Comment by nemchik** [comment]
So question about this. If I have multiple tabs playing and I press Pause, from what I can tell by reading your changes, all the playing tabs will Pause. Does one of those tabs get marked as the one that was most recently playing? Or do all of them? When I press Play what happens? Will all of the tabs that were playing start playing again? Or only one?

By the way I really like what you've got here, I just want to know how it works :) If I have time I might try too cherry pick and test it as well.

**Comment by metaflow** [comment]
@nemchik If there are several tabs playing then yes, action "play / pause", "next" etc. will be applied to all of them. It's not deterministic which one will be picked as "recent" when  you press "play" though it will be only one. I can probably improve that heuristic by checking if there are several tabs with "close enough" update timestamp (say 200 ms) then 1. pick active tab if any 2. pick most recent one (as right now).

**Comment by Wizek** [comment]
@metaflow So, I've installed your branch, but haven't yet been able to give it a thorough test yet. But one thing I noticed is that when I have netflix in an chrome app window instead of a tab (created via: vertical 3 dots in chrome upper right corenr ->more tools -> Add to desktop... ) it doesn't seem to register as most recently played. Could it be that such app windows are not yet working? And if so, maybe it can be as simple as adding another event listener along the lines of listening for window focus changes and not just tab focus changes?

**Comment by nemchik** [comment]
More related issues that I believe this would solve:
https://github.com/berrberr/streamkeys/issues/264
https://github.com/berrberr/streamkeys/issues/243
Possibly https://github.com/berrberr/streamkeys/issues/206
https://github.com/berrberr/streamkeys/issues/193
Possibly https://github.com/berrberr/streamkeys/issues/192
https://github.com/berrberr/streamkeys/issues/133

And with a few more changes/additions possibly https://github.com/berrberr/streamkeys/issues/58 (maybe have an option in the settings to allow newly opened tabs to pause all playing tabs, but that might deserve a separate PR after this one).

**Comment by metaflow** [comment]
@Wizek could you please provide more details for that, i.g. how to reproduce that? I have tried to do the following: create desktop "app" to youtube. Then I opened youtube page in "normal mode" and paused it. Opened app and started a playback. Pressing play/pause interacted with "app" as one would expect.
One thing to watch for is to make sure that both windows belong to the same user profile.

**Comment by metaflow** [comment]
@nemchik I went through the list of the issues above
#264 - slightly different requirement. The author wants to explicitly interact with active tab only but "most recent" might work for him as well (asked).
#243 - yes
Possibly #206 - you can do that by pressing "stop" key thought there is no UI element to do that
#193 - yes
Possibly #192 - no, that one is about ordering of the players in popup
#133 - yes, though I have not read all comments

**Comment by metaflow** [comment]
@nemchik I wonder if it's OK to wait for a while for @berrberr or should I fix something?

**Comment by nemchik** [comment]
I think this looks good. I would wait to see if @berrberr has any input. If it gets merged then that would be a good time to look into additional PRs that might depend on this code.

**Comment by berrberr** [comment]
@metaflow thank you so much for this! Really like the idea. I am going to find some time today to review this, then merge in if ready and add to next extension version.

**Comment by nemchik** [comment]
BTW I believe I actually did end up cherry picking this PR and testing it out and it worked well on all sites I tried while having multiple tabs with players open.

**Comment by berrberr** [comment]
Awesome, thank you very much @metaflow 

I tested and reviewed, everything appears to be good.

This will be included in the next version. I think that this should be a very useful feature to some people.

Thanks!

**Comment by metaflow** [comment]
Thank you @berrberr ! Glad you found this useful.

**Comment by metaflow** [comment]
are you going to close some of the related issues mentioned by @nemchik ?

**Comment by natebot13** [comment]
So I was thinking that it might be useful to utilize tab order if that's a useful heuristic? It's much easier to edit tab order than it is to go into the settings and modify the priority, and I'm pretty much always only listening to only one thing at a time anyway. This also solves the issue with having the same sites open in multiple tabs, (i.e. multiple YouTube videos) The "Last Used" feature seems a little difficult to figure out if the user presses play on an silent chrome window with a bunch of media tabs open.
Perhaps an option to enable priority based on tab order would work?

**Pull request merged**